### PR TITLE
Display proposal details in a modal instead of tooltip

### LINKED ISSUE #822: Info icon inconsistent between mouse over or click
The info icons are not consistently implemented. Some require click and some are mouse-over. Fix the new governance pages to mouse-over for consistency.

Screen Shot 2022-03-23 at 11 11 32 AM

### PR DESCRIPTION
**Pull request by AdimekweEbuka**
Resolve #822 

- Change info icon to clickable icon button
- Replace info tooltip with info modal
- Add title "Proposal vote details" to votebar info modal 

godcr (testnet) 2022-04-26 12-37-01

### CODE REVIEW

**Review by devchoplife — CHANGES_REQUESTED** [review]
The info icon should be on the same line with the vote options as it was initially. Also, it should be aligned right (maintain the same line with the progress bar)
Screenshot from 2022-04-24 22-55-52

File: ui/page/components/votebar_widget.go [hunk 1]
```diff
@@ -253,68 +274,68 @@ func (v *VoteBar) layoutIconAndText(gtx C, lbl decredmaterial.Label, count float
...
+// 	return dims
+// }
+
+// func (v *VoteBar) layoutInfoTooltip(gtx C, rect image.Rectangle) {
+// 	inset := layout.Inset{Top: unit.Dp(20), Left: unit.Dp(-180)}
+// 	col := v.Theme.Color.GrayText2
+
+// 	v.quorumTooltip.Layout(gtx, rect, inset, func(gtx C) D {
+// 		gtx.Constraints.Max.X = gtx.Px(unit.Dp(180))
+// 		gtx.Constraints.Min.X = gtx.Constraints.Max.X
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]: If these are no longer needed, lets remove them.

File: ui/page/components/votebar_widget.go [hunk 2]
```diff
@@ -226,6 +233,20 @@ func (v *VoteBar) Layout(gtx C) D {
...
 
+func (v *VoteBar) infoButtonModal() {
+	text1 := fmt.Sprintf("Total votes:  %6.0f", v.totalVotes)
+	text2 := fmt.Sprintf("Quorum requirement:  %6.0f", (v.requiredPercentage/100)*v.eligibleVotes)
+	text3 := fmt.Sprintf("Discussions:   %d comments", v.numComment)
+	text4 := fmt.Sprintf("Published:   %s", dcrlibwallet.FormatUTCTime(v.publishedAt))
+	text5 := fmt.Sprintf("Token:   %s", v.token)
+	bodyText := fmt.Sprintf("%s\n %v\n %s\n %s\n %s", text1, text2, text3, text4, text5)
+	modal.NewInfoModal(v.Load).
+		Title("").
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]: Title should be the vote title.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image

Kindly revert the changes that added the padding below and removed the text in front of the info icon. See original page info.

image

This issue/Pr is meant to only move the tooltip info to a modal.

File: ui/page/components/votebar_widget.go [hunk 3]
```diff
@@ -226,6 +227,20 @@ func (v *VoteBar) Layout(gtx C) D {
...
 
+func (v *VoteBar) infoButtonModal() {
+	text1 := fmt.Sprintf("Total votes:  %6.0f", v.totalVotes)
+	text2 := fmt.Sprintf("Quorum requirement:  %6.0f", (v.requiredPercentage/100)*v.eligibleVotes)
+	text3 := fmt.Sprintf("Discussions:   %d comments", v.numComment)
+	text4 := fmt.Sprintf("Published:   %s", dcrlibwallet.FormatUTCTime(v.publishedAt))
+	text5 := fmt.Sprintf("Token:   %s", v.token)
+	bodyText := fmt.Sprintf("%s\n %v\n %s\n %s\n %s", text1, text2, text3, text4, text5)
+	modal.NewInfoModal(v.Load).
+		Title("Vote title").
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]: pass the proposal title to the vote bar info button and use the proposal title.

Better still just use `Proposal vote details`

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Comment by dreacot** [comment]
please add the proper screenshots

**Comment by AdimekweEbuka** [comment]
> please add the proper screenshots

Noted, will add now.

**Comment by AdimekweEbuka** [comment]
godcr (testnet) 2022-04-26 12-37-01


> > please add the proper screenshots
> 
> Noted, will add now.
godcr (testnet) 2022-04-26 12-37-01

**Comment by AdimekweEbuka** [comment]
> > please add the proper screenshots
> 
> Noted, will add now.
godcr (testnet) 2022-04-26 12-37-01

**Comment by dreacot** [comment]
@AdimekweEbuka edit the original description and add it there

File: ui/page/components/votebar_widget.go [hunk 4]
```diff
@@ -55,10 +56,10 @@ func NewVoteBar(l *load.Load) *VoteBar {
 		noColor:       l.Theme.Color.Danger,
 		passTooltip:   l.Theme.Tooltip(),
 		quorumTooltip: l.Theme.Tooltip(),
-		infoIcon:      decredmaterial.NewIcon(l.Theme.Icons.ActionInfo),
 		legendIcon:    decredmaterial.NewIcon(l.Theme.Icons.ImageBrightness1),
+		newInfoButton: l.Theme.IconButton(l.Theme.Icons.ActionInfo),
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]:
```suggestion
```

File: ui/page/components/votebar_widget.go [hunk 5]
```diff
@@ -55,10 +56,10 @@ func NewVoteBar(l *load.Load) *VoteBar {
...
 	}
-	vb.infoIcon.Color = l.Theme.Color.Gray1
+	vb.newInfoButton.Inset = layout.Inset{}
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]:
```suggestion
_, pg.infoButton = components.SubpageHeaderButtons(l)
	pg.infoButton.Size = values.MarginPadding20
```

File: ui/page/components/votebar_widget.go [hunk 6]
```diff
@@ -41,8 +42,8 @@ type VoteBar struct {
 	passTooltip   *decredmaterial.Tooltip
 	quorumTooltip *decredmaterial.Tooltip
 
-	infoIcon   *decredmaterial.Icon
-	legendIcon *decredmaterial.Icon
+	legendIcon    *decredmaterial.Icon
+	newInfoButton decredmaterial.IconButton
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]:
```suggestion
	infoButton decredmaterial.IconButton
```

File: ui/page/components/votebar_widget.go [hunk 7]
```diff
@@ -257,64 +272,14 @@ func (v *VoteBar) layoutInfo(gtx C) D {
...
-				Min: gtx.Constraints.Min,
-				Max: gtx.Constraints.Max,
+			if v.newInfoButton.Button.Clicked() {
+				v.infoButtonModal()
 			}
-			rect.Max.Y = 20
-			v.layoutInfoTooltip(gtx, rect)
 			return layout.Inset{Left: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
-				return v.infoIcon.Layout(gtx, values.MarginPadding20)
+				return v.newInfoButton.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/components/votebar_widget.go]:
```suggestion
				return v.infoButton.Layout(gtx)
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**